Services/Administration/OJS.Services.Administration.Business/Participants/ParticipantsBusinessService.cs

```diff
@@ -93,10 +94,22 @@ public async Task<string> UpdateParticipationTimeForMultipleParticipants(
 
         var contestDurationInMinutes = contest!.Duration!.Value.Minutes;
 
-        var participantsInTimeRange = await this.participantsData
-            .All(p => p.ParticipationStartTime >= model.ChangeParticipationTimeRangeStart &&
-                               p.ParticipationStartTime <= model.ChangeParticipationTimeRangeEnd &&
-                               p.Contest.Type == ContestType.OnlinePracticalExam);
+        var invalidParticipantsUsernames = await this.participantsData
+            .GetAllOfficialInOnlineContestByContestAndParticipationStartTimeRange(
+                model.ContestId,
+                (DateTime)model.ChangeParticipationTimeRangeStart,
+                (DateTime)model.ChangeParticipationTimeRangeEnd)
+            .Where(p => p.ParticipationEndTime.HasValue && p.ParticipationStartTime.HasValue &&
+                        p.ParticipationEndTime.Value.AddMinutes(model.TimeInMinutes) <
+                        p.ParticipationStartTime.Value.AddMinutes(contestDurationInMinutes))
+            .Select(p => p.User.UserName)
+            .ToListAsync();
```

Comment:
You can filter these usernames from the `participantsInTimeRange` below, no need to make an extra request.

Reply:
These are two separate requests because the `.Where()` filters are different.

Reply:
Done.